src/components/subsidy-request-configuration/data/hooks.js

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'react';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import { logError } from '@edx/frontend-platform/logging';
+import EnterpriseAccessApiService from '../../../data/services/EnterpriseAccessApiService';
+import EcommerceApiService from '../../../data/services/EcommerceApiService';
+import LicenseManagerApiService from '../../../data/services/LicenseManagerAPIService';
+import { SUPPORTED_SUBSIDY_TYPES } from '../../../data/constants/subsidyRequests';
+
+/**
+ * Fetches the subsidy request configuration for the enterprise. If not found, creates a new
+ * configuration with the correct subsidy type used for requests.
+ *
+ * @param {string} enterpriseUUID UUID of the enterprise
+ * @returns {Object} {customerConfiguration: Object, isLoading: bool}
+ */
+// eslint-disable-next-line import/prefer-default-export
+export const useSubsidyRequestConfiguration = (enterpriseUUID) => {
+  const [subsidyRequestConfiguration, setSubsidyRequestConfiguration] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (!enterpriseUUID) {
+      return;
+    }
+
+    const createCustomerConfiguration = async () => {
+      try {
+        const [couponsData, subscriptionsData] = await Promise.all([
+          EcommerceApiService.fetchCouponOrders(),
+          LicenseManagerApiService.fetchSubscriptions({
+            enterprise_customer_uuid: enterpriseUUID,
+          }),
+        ]);
+        const hasCoupons = couponsData.data.results.length > 0;
+        const hasSubscriptions = subscriptionsData.data.results.length > 0;
+
+        // If the customer has two subsidy types, they are not eligible for Browse & Request in the MVP
+        // A subsidy type of null on the customer configuration indicates that the customer is not eligible
+        let subsidyType = null;
+
+        if (!(hasCoupons && hasSubscriptions)) {
+          if (hasCoupons) {
+            subsidyType = SUPPORTED_SUBSIDY_TYPES.coupon;
+          } else if (hasSubscriptions) {
+            subsidyType = SUPPORTED_SUBSIDY_TYPES.license;
+          }
+        }
+
+        const response = await EnterpriseAccessApiService.createSubsidyRequestConfiguration({
+          enterpriseId: enterpriseUUID,
+          subsidyType,
```

Comment:
Never mind, this question can be ignored 😄  This is what's expected on the backend side of things (i.e., `subsidy_type` is a nullable field).

Reply:
The field is nullable on the configuration, and we're using it to represent eligibility. If the field is non-null, we know the enterprise can use b&r for that subsidy type, otherwise, they are ineligible and should not see any b&r related features.